autogen/code_utils.py

```diff
@@ -269,10 +280,37 @@ def execute_code(
     in_docker_container = os.path.exists("/.dockerenv")
     if not use_docker or in_docker_container:
         # already running in a docker container
-        cmd = [
-            sys.executable if lang.startswith("python") else _cmd(lang),
-            f".\\{filename}" if WIN32 else filename,
-        ]
+        cmd_lang = _cmd(lang)
+        if cmd_lang == 'tsc':
+            # TypeScript file
+            cmd = ['tsc', filename]
```

Comment:
consider using `tsx`, it may go smoother than typescript compiler with all the possible configuration issues
https://www.npmjs.com/package/tsx